modules/junit-jupiter/src/main/java/org/testcontainers/junit/jupiter/TestcontainersExtension.java

```diff
@@ -99,17 +100,16 @@ private void signalAfterTestToContainersFor(String storeKey, ExtensionContext co
     }
 
     private TestDescription testDescriptionFrom(ExtensionContext context) {
-        return new TestDescription() {
-            @Override
-            public String getTestId() {
-                return context.getUniqueId();
-            }
+        return new TestcontainersTestDescription(
+            context.getUniqueId(),
+            filesystemFriendlyNameOf(context)
+        );
+    }
 
-            @Override
-            public String getFilesystemFriendlyName() {
-                return context.getDisplayName();
-            }
-        };
+    private String filesystemFriendlyNameOf(ExtensionContext context) {
+        return context.getRequiredTestClass().getName()
+            + "-"
+            + context.getTestMethod().map(Method::getName).orElse("static");
```

Comment:
@bsideup @rnorth I've changed how the filesystem friendly name is generated. However, I'm not quite sure if this is a good way.
WDYT?

Reply:
what is "static" here? Also, I believe `getRequiredTestClass` will fail if dynamic tests are used